src/main/kotlin/racingcar/NamedPosition.kt

```diff
@@ -0,0 +1,24 @@
+package racingcar
+
+interface NamedPosition {
+    val name: Name
+    fun intValue(): Int
+    operator fun compareTo(other: Int): Int
+    operator fun inc(): NamedPosition
+
+    data class Smart(override val name: Name, private val position: Int = 0) : NamedPosition {
+        init {
+            require(position >= 0)
+        }
+
+        override operator fun inc(): Smart {
+            return copy(position = position + 1)
+        }
+
+        override fun intValue(): Int {
+            return position
+        }
+
+        override operator fun compareTo(other: Int) = position - other
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        override operator fun compareTo(other: Int) = position.compareTo(other)
```

Reply:
이게 더 좋겠네요!
72f6403256bfde0d73b2e30ef8762d4c2481502d 에서 적용하였습니다.